hail/src/test/scala/is/hail/expr/ir/MatrixIRSuite.scala

```diff
@@ -273,4 +273,35 @@ class MatrixIRSuite extends SparkSuite {
       assert(values.isSorted && values.length == 11, n -> shuffle)
     }
   }
+
+  @Test def testMatrixWrite() {
+    // Native Writer
+    val range = MatrixTable.range(hc, 11, 3, Some(10))
+    var path = tmpDir.createLocalTempFile(extension = "mt")
+    Interpret(MatrixWrite(range.ast, MatrixNativeWriter(path)))
+    val read = MatrixTable.read(hc, path)
+    assert(read.same(range))
```

Comment:
Let's use four separate test methods so we have more fine-grained test success/failure information. you can make the `MatrixTable.range...` a class-level `lazy val`.